### ISSUE TITLE
Implement the rest of the commands

### ISSUE THREAD

**Issue by craftbyte** [applest/node-applest-atem#23]
It would be nice if the plugin implemented all of the features of the ATEM protocol. Most of them are listed here: http://skaarhoj.com/fileadmin/BMDPROTOCOL.html

**Comment by filiphanes** [comment]
Hi, which features do you miss?

**Comment by auto456** [comment]
Macro calling would be really nice.
Is it possible?

**Comment by johnnydanger** [comment]
It’s already in there: runMacro(number) should do it.  If you go into the applest-atem module folder and look at the atem.js code you’ll see all the functions that have been implemented.  There are a number of macro related functions in place.

Cheers.



–––––
John Meloy
john@johnnydanger.net
phone (360) 201-9039
http://www.johnnydanger.net
http://danger.photos

**Comment by auto456** [comment]
Yee, sorry I was to late finding it and deleted my Question.

But is there a way to add other functions by myself? Can I find the Hex Code(?) reference anywhere for the commands from the Blackmagic API? 
For example to change the supersource as I can do with the mixer itself or through the XML files.

**Comment by johnnydanger** [comment]
Absolutely.  You can find all the reverse engineered ATEM switcher protocol info here:

https://www.skaarhoj.com/fileadmin/BMDPROTOCOL.html

You’d want to look at the CSBP super source box parameters set string info, or the CSSc super source parameters.




–––––
John Meloy
john@johnnydanger.net
phone (360) 201-9039
http://www.johnnydanger.net
http://danger.photos
